boa/src/builtins/set/set_iterator.rs

```diff
@@ -0,0 +1,149 @@
+use crate::{
+    builtins::function::make_builtin_fn,
+    builtins::iterable::create_iter_result_object,
+    builtins::Array,
+    builtins::Value,
+    object::{GcObject, ObjectData},
+    property::{Attribute, DataDescriptor},
+    BoaProfiler, Context, Result,
+};
+use gc::{Finalize, Trace};
+
+#[derive(Debug, Clone, Finalize, Trace)]
+pub enum SetIterationKind {
+    Value,
+    KeyAndValue,
+}
+
+/// The Set Iterator object represents an iteration over a set. It implements the iterator protocol.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-set-iterator-objects
+#[derive(Debug, Clone, Finalize, Trace)]
+pub struct SetIterator {
+    iterated_set: Value,
+    set_next_index: usize,
+    set_iteration_kind: SetIterationKind,
```

Comment:
```suggestion
    next_index: usize,
    iteration_kind: SetIterationKind,
```
I think it's clearer if we don't prefix the fields with `set_` since they are already in the `SetInterator`.